src/boss/mar/AFC4.c

```diff
@@ -11,14 +11,640 @@ extern Dialogue g_Dialogue;
 
 #include "../../st/cutscene_unk4.h"
 
-INCLUDE_ASM("boss/mar/nonmatchings/AFC4", func_us_8018B2F0);
+extern char* D_us_80180680[];
+
+void func_us_8018B2F0(u16 actorIndex, Entity* self) {
+    Primitive* prim;
+    s16 primIndex;
+    s32 x;
+    u16 chCount;
+    const char* actorName;
+    char ch;
+
+    actorName = D_us_80180680[actorIndex];
+    chCount = 0;
+    while (true) {
+        ch = *actorName++;
+        if (ch == DIAG_EOL) {
+            ch = *actorName++;
+            if (ch == DIAG_EOS) {
+                break;
+            }
+        }
+        if (ch == MENUCHAR(' ')) {
+            continue;
+        }
+        chCount++;
+    }
+
+    // Create chCount amount of sprites based on the actor name's letter count
+    primIndex = g_api.AllocPrimitives(PRIM_SPRT, chCount);
+    if (primIndex == -1) {
+        DestroyEntity(self);
+        return;
+    }
+
+    // Fill prims to render the actor name on screen
+    prim = &g_PrimBuf[primIndex];
+    g_Dialogue.primIndex[1] = primIndex;
+    actorName = D_us_80180680[actorIndex];
+    x = 0x38;
+    while (prim != NULL) {
+        ch = *actorName++;
+        if (ch == MENUCHAR(' ')) {
+            x += FONT_SPACE;
+        } else {
+            prim->type = PRIM_SPRT;
+            prim->tpage = 0x1E;
+            prim->clut = 0x196;
+            prim->u0 = (ch & 0x0F) * FONT_W;
+            prim->v0 = (ch & 0xF0) / (FONT_H / 4);
+            prim->v1 = FONT_H;
+            prim->u1 = FONT_W;
+            prim->priority = 0x1FF;
+            prim->drawMode = DRAW_HIDE;
+            prim->x0 = x;
+            prim->y0 = g_Dialogue.startY + 6;
+            prim = prim->next;
+            x += FONT_GAP;
+        }
+    }
+}
 
 #include "../../st/cutscene_unk6.h"
 
 #include "../../st/cutscene.h"
 
-INCLUDE_ASM("boss/mar/nonmatchings/AFC4", func_us_8018B74C);
+// Animates the portrait size of the actor by enlarging or shrinking it
+void func_us_8018B74C(u8 ySteps) {
```

Comment:
This is a duplicate, could we factor it out? Can wait for a followup PR

Reply:
I'll take a look at these.